### PR TITLE
Github4519

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
Fixed 4519

#### What does this implement/fix? Explain your changes.
This fixes 2 bugs.
The first is the one reported by @rachelnwalker about poor placement of SGroup data in a grid picture.  This turned out to be a consequence of drawMolecules() calling setupDrawMolecule twice for each molecule.
The second was the poor placement of SGroupData due to using TestAlignType::START for Group data.

#### Any other comments?
Sorry it's taken so long to get round to fixing these.  The grid one was a real pain both to track down and fix.

### CODE REVIEW

**Comment by DavidACosgrove** [comment]
The failures are due to one of the test drawings (radicals in a grid) being smaller than previously.  I'll take a look to see if that's reasonable or not.

**Comment by DavidACosgrove** [comment]
The new drawing of radicals in a grid is scaled slightly larger than the current master, which is consistent with the bug fix so I've updated the expected test values.

File: Code/GraphMol/MolDraw2D/DrawText.h [hunk 1]
```diff
@@ -55,7 +55,7 @@ class RDKIT_MOLDRAW2D_EXPORT DrawText {
   double minFontSize() const;
   void setMinFontSize(double new_max);
   double fontScale() const;
-  void setFontScale(double new_scale);
+  void setFontScale(double new_scale, bool ignore_extremes=false);
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/DrawText.h:58]:
```suggestion
  void setFontScale(double new_scale, bool ignoreExtremes=false);
```

File: Code/GraphMol/MolDraw2D/DrawText.cpp [hunk 2]
```diff
@@ -64,14 +64,18 @@ void DrawText::setMinFontSize(double new_min) { min_font_size_ = new_min; }
 double DrawText::fontScale() const { return font_scale_; }
 
 // ****************************************************************************
-void DrawText::setFontScale(double new_scale) {
+void DrawText::setFontScale(double new_scale, bool ignore_extremes) {
   font_scale_ = new_scale;
+  if (ignore_extremes) {
+    return;
+  }
+
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/DrawText.cpp:72]:
```suggestion
void DrawText::setFontScale(double new_scale, bool ignoreExtremes) {
  font_scale_ = new_scale;
  if (ignoreExtremes) {
    return;
  }

```

Comment (greglandrum) [reply Code/GraphMol/MolDraw2D/DrawText.cpp:72]: minor change to try and avoid having the API get even more inconsistent in terms of naming

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp [hunk 3]
```diff
@@ -2406,14 +2407,13 @@ void MolDraw2D::extractAtomNotes(const ROMol &mol) {
...
     if (atom->getPropIfPresent(common_properties::atomNote, note)) {
       if (!note.empty()) {
-        auto note_rect = calcAnnotationPosition(mol, atom, note);
-        if (note_rect.width_ < 0.0) {
+        AnnotationType annot;
+        annot.text_ = note;
+        calcAnnotationPosition(mol, atom, annot);
+        if (annot.rect_.width_ < 0.0) {
           cerr << "Couldn't find good place for note " << note << " for atom "
                << atom->getIdx() << endl;
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/MolDraw2D.cpp:2415]: I know this isn't part of your PR, but as long as you're in here, please change this to:
```suggestion
          BOOST_LOG(rdWarningLog) << "Couldn't find good place for note " << note << " for atom "
               << atom->getIdx() << std::endl;
```

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp [hunk 4]
```diff
@@ -2440,15 +2440,14 @@ void MolDraw2D::extractMolNotes(const ROMol &mol) {
...
   if (!note.empty()) {
-    auto note_rect = calcAnnotationPosition(mol, note);
-    if (note_rect.width_ < 0.0) {
+    AnnotationType annot;
+    annot.text_ = note;
+    annot.align_ = TextAlignType::START;
+    annot.scaleText_ = false;
+    calcAnnotationPosition(mol, annot);
+    if (annot.rect_.width_ < 0.0) {
       cerr << "Couldn't find good place for molecule note " << note << endl;
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/MolDraw2D.cpp:2449]: again, not part of this PR, but:
```suggestion
      BOOST_LOG(rdWarningLog) << "Couldn't find good place for molecule note " << note << std::endl;
```

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp [hunk 5]
```diff
@@ -2464,14 +2463,13 @@ void MolDraw2D::extractBondNotes(const ROMol &mol) {
...
     if (bond->getPropIfPresent(common_properties::bondNote, note)) {
       if (!note.empty()) {
-        auto note_rect = calcAnnotationPosition(mol, bond, note);
-        if (note_rect.width_ < 0.0) {
+        AnnotationType annot;
+        annot.text_ = note;
+        calcAnnotationPosition(mol, bond, annot);
+        if (annot.rect_.width_ < 0.0) {
           cerr << "Couldn't find good place for note " << note << " for bond "
                << bond->getIdx() << endl;
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/MolDraw2D.cpp:2471]: Again, not part of this PR, but:
```suggestion
          BOOST_LOG(rdWarningLog) << "Couldn't find good place for note " << note << " for bond "
               << bond->getIdx() << std::endl;
```

**Review by greglandrum — CHANGES_REQUESTED** [review]
Mostly looks good.
My suggested changes are all small cleanup things.

**Review by greglandrum — APPROVED** [review]

**Pull request merged**